src/components/MarcAuthorityView/MarcAuthorityView.js

```diff
@@ -0,0 +1,126 @@
+import { useContext } from 'react';
+import { useIntl } from 'react-intl';
+import { useQueryClient } from 'react-query';
+import PropTypes from 'prop-types';
+
+import { Loading } from '@folio/stripes/components';
+import {
+  useMarcSource,
+  useAuthority,
+  QUERY_KEY_AUTHORITY_SOURCE,
+  SelectedAuthorityRecordContext,
+} from '@folio/stripes-authority-components';
+import {
+  useCallout,
+  useNamespace,
+} from '@folio/stripes/core';
+import MarcView from '@folio/quick-marc/src/QuickMarcView/QuickMarcView';
+
+import cloneDeep from 'lodash/cloneDeep';
+import set from 'lodash/set';
+import { MAIN_PANE_HEIGHT } from '../../constants';
+
+const propTypes = {
+  onCloseDetailView: PropTypes.func.isRequired,
+};
+
+const MarcAuthorityView = ({
+  onCloseDetailView,
+}) => {
+  const queryClient = useQueryClient();
+  const authoritySourceNamespace = useNamespace({ key: QUERY_KEY_AUTHORITY_SOURCE });
+  const callout = useCallout();
+  const intl = useIntl();
+  const [selectedAuthorityRecordContext] = useContext(SelectedAuthorityRecordContext);
+  const { id, authRefType, headingRef } = selectedAuthorityRecordContext;
+
+  const markHighlightedFields = (marcSource, authority) => {
+    const highlightAuthRefFields = {
+      'Authorized': /1\d\d/,
+      'Reference': /4\d\d/,
+      'Auth/Ref': /5\d\d/,
+    };
+
+    const marcFields = marcSource.data.parsedRecord.content.fields.map(field => {
```

Comment:
Here's a lot of duplication from authorities app - please create a util function in stripes-authority-components so we can reuse it in plugin and app

Reply:
Thanks, fixed